work/worker.go

```diff
@@ -538,9 +538,10 @@ func (self *worker) commitNewWork() {
 	// Create the current work task
 	work := self.current
 	if self.nodetype == common.CONSENSUSNODE {
-		// TODO : Need to KIP-71 hardfork.
-		// TODO : It need to uncomment after implemented baseFee logic.
-		// pending = types.FilterTransactionWithBaseFee(pending, baseFee)
+		if self.config.IsKIP71ForkEnabled(header.Number) {
+			header.BaseFee = misc.NextBlockBaseFee(parent.Header(), self.config)
+			pending = types.FilterTransactionWithBaseFee(pending, header.BaseFee)
+		}
```

Comment:
`FilterTransactionWithBaseFee()` function has a heavy nested for-loop.
I think we should run heavy tasks before waiting for ideal time to not consume precious consensus time.
Can we move this if block to the start of `commitNewWork`? (where we've just obtained `pending` list)

Reply:
Thanks for good Idea. I'll try it

Reply:
On a second thought, we might be able to integrate FilterTransactionWithBaseFee into Pending(). Let me elaborate in another comment.

Reply:
Yeah, @blukat29 actually I also found many sites to be refactored. However I'd like to remain it `TODO`.
Because I think txPool code is complex and some fix can cause unexpected side effects. How about considering it separately later?

Reply:
That's fair. I see that performance is not the top priority.